src/Planners/src/QuinticSpline.cpp

```diff
@@ -0,0 +1,787 @@
+                                                 const int& rowOffset,
+                                                 const int& columnOffset,
+                                                 std::vector<Eigen::Triplet<double>>& tripletList)
+{
+    const auto& poly = polynomials;
+
+    // The following triplets represent this matrix
+    // /    8          1    \
+    // | ------     ------  |
+    // |  2                 |
+    // | T           T      |
+    // |  i - 1       i - 1 |
+    // |                    |
+    // |                    |
+    // |   14          2    |
+    // | ------     ------  |
+    // |  3          2      |
+    // | T          T       |
+    // \  i - 1      i - 1  /
+
+    tripletList.push_back({rowOffset, columnOffset, 8 / std::pow(poly[knotIndex - 1].duration, 2)});
+    tripletList.push_back({rowOffset, columnOffset + 1, 1 / poly[knotIndex - 1].duration});
+    tripletList.push_back(
+        {rowOffset + 1, columnOffset, 14 / std::pow(poly[knotIndex - 1].duration, 3)});
+    tripletList.push_back(
+        {rowOffset + 1, columnOffset + 1, 2 / std::pow(poly[knotIndex - 1].duration, 2)});
+}
+
+void QuinticSpline::Impl::addTripletNextKnot(const int& knotIndex,
+                                             const int& rowOffset,
+                                             const int& columnOffset,
+                                             std::vector<Eigen::Triplet<double>>& tripletList)
+{
+    const auto& poly = polynomials;
+
+    // The following triplets represent this matrix
+    // /    8          1    \
+    // | - ---        ---   |
+    // |    2               |
+    // |   T           T    |
+    // |    i           i   |
+    // |                    |
+    // |                    |
+    // |  14           2    |
+    // |  --         - ---  |
+    // |   3            2   |
+    // |  T            T    |
+    // \   i            i   /
+
+    tripletList.push_back({rowOffset, columnOffset, -8 / std::pow(poly[knotIndex].duration, 2)});
+    tripletList.push_back({rowOffset, columnOffset + 1, 1 / poly[knotIndex].duration});
+    tripletList.push_back(
+        {rowOffset + 1, columnOffset, 14 / std::pow(poly[knotIndex].duration, 3)});
+    tripletList.push_back(
+        {rowOffset + 1, columnOffset + 1, -2 / std::pow(poly[knotIndex].duration, 2)});
+}
+
+void QuinticSpline::Impl::addKnownTermKnotPosition(const std::size_t& knotIndex,
+                                             const std::size_t& coordinateIndex,
+                                             Eigen::Ref<Eigen::VectorXd> b)
+{
+    const auto& poly = polynomials;
+
+    const auto& i = knotIndex;
+    const auto& j = coordinateIndex;
+
+    b[0] += 20
+            * (1 / std::pow(poly[i - 1].duration, 3)
+                   * (knots[i].position[j] - knots[i - 1].position[j])
+               + 1 / std::pow(poly[i].duration, 3)
+                     * (knots[i].position[j] - knots[i + 1].position[j]));
```

Comment:
Are you using a `+=` here assuming that the existing contents are zero?

Reply:
Yep, `addKnownTermKnotPosition` is called only in this part of the code and here I reset to zero the vector every time 
https://github.com/dic-iit/bipedal-locomotion-framework/blob/97ebb0bfe8ea740c1318b2601865ae869e0ca41c/src/Planners/src/QuinticSpline.cpp#L618-L644

Reply:
Understood.